src/validation/entities/schemas/timezone.schema.ts

```diff
@@ -0,0 +1,40 @@
+import { z } from 'zod';
+
+/**
+ * Validates a timezone
+ *
+ * @param {string | undefined} timezone The timezone string to check for validity
+ *
+ * @returns {boolean} Returns 'true' if the timezone is valid, otherwise 'false'
+ */
+export const TimezoneSchema = z
+  .string()
+  .optional()
+  .refine(
+    (timezone: string | undefined): boolean => {
+      if (timezone) {
+        if (!isTimezoneEnabled() || !isTimezoneValid(timezone)) {
+          return false;
+        }
+      }
+
+      return true;
+    },
+    { message: 'Invalid Timezone' },
+  );
+
+const isTimezoneEnabled = (): boolean => {
+  return !Intl || !Intl.DateTimeFormat().resolvedOptions().timeZone
+    ? false
+    : true;
+};
```

Comment:
We aren't checking for `Intl` support elsewhere so I think we can safely remove the initial check, then simplify the rest, e.g.

```suggestion
const isTimezoneEnabled = (): boolean => {
  return !!Intl.DateTimeFormat().resolvedOptions().timeZone
};
```

However, considering support is in place - do we need this function?

Reply:
I agree, I think `Intl` should be always supported, right? It seems it is supported since Node 0.12 (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl)

Reply:
Personally, I find it confusing to have two `!!`. I believe an explicit true, false is way more clear.
 I would say let's keep `Intl` check as well since otherwise if for whatever reason `Intl` is disabled we will get an error instead of `true` or `false`. What do you think?

Reply:
@hectorgomezv It depends. You can disabled `Intl` when building node binary. more on that:
https://nodejs.org/api/intl.html#options-for-building-nodejs
But the good thing is that the binaries in docker hub have the default behavior i.e. full `Intl` support.
https://github.com/nodejs/docker-node/blob/0c0069246367ac5ac0fc6bca141fb04faaca2f4b/22/alpine3.19/Dockerfile

Actually, that's a very strict check.

Reply:
I'm impartial regarding the ternary.

If we are trying to be defensive regarding `Intl`, shouldn't we also add the check everywhere it's used?

Reply:
Oh, I didn't know that, thanks!  Then I'd leave this up to you. 😄